docs/proposals/synchronous-garbage-collection.md

```diff
@@ -56,18 +57,33 @@ DeleteOptions {
 }
 ```
 
+**OwnerReference**
+```go
+OwnerReference {
+    ...
+    // Should the owner be deleted from the key-value store after this object is deleted during synchronous garbage collection.
+    // If the user creates the OwnerReference has delete permission of the owner, BlockSynchronousGC defaults to true; otherwise the field defaults to false, also 422 will be returned if the field is set to false.
+    BlockSynchronousGC *bool
```

Comment:
`Synchronize` as a name for this. And I think this is an alternative design. You wouldn't do this AND the finalizer, it's one or the other. The idea with this is that the links are auth checked at admission time rather than GC time.

Actually I think you should add a `DeletionStrategy` field, with "background" (default) or "synchronous" as the possible values. It's bad to just drop bools in the API.